R/helpers_ggbetween_anova_subtitles.R

```diff
@@ -646,3 +646,155 @@ subtitle_anova_bayes <- function(data,
+#'   (Default: `k = 2`).
+#' @param messages Decides whether messages references, notes, and warnings
+#'   are to be displayed (Default: `TRUE`).
+#' @param ... Additional arguments.
+#' @inheritParams stats::oneway.test
+#' @inheritParams subtitle_t_parametric
+#' @inheritParams lm_effsize_standardizer
+#'
+#' @importFrom ez ezANOVA
+#' @importFrom MBESS conf.limits.ncf
+#' @importFrom dplyr select
+#' @importFrom rlang !! enquo
+#' @importFrom sjstats eta_sq omega_sq
+#'
+#' @examples
+#' # setup
+#' set.seed(123)
+#' library(ggstatsplot)
+#' library(jmv)
+#' data("bugs", package = "jmv")
+#'
+#' # converting to long format
+#' data_bugs <- bugs %>%
+#'   dplyr::filter(., !is.na(LDHF) & !is.na(LDLF) & !is.na(HDLF)  & !is.na(HDHF)) %>%
+#'   tibble::as_tibble(.) %>%
+#'   tidyr::gather(., key, value, LDLF:HDHF)
+#'
+#' # creating the subtitle
+#' ggstatsplot::subtitle_anova_parametric_repeated(
+#'   data = data_bugs,
+#'   x = key,
+#'   y = value,
+#'   id.variable = Subject,
+#'   k = 2
+#' )
+#'
+#' @export
+
+# function body
+subtitle_anova_parametric_repeated <- function(data,
+                                      x,
+                                      y,
+                                      id.variable,
+                                      effsize.type = "unbiased",
+                                      conf.level = 0.95,
+                                      k = 2,
+                                      messages = TRUE,
+                                      ...) {
+
+  # creating a dataframe
+  data <-
+    dplyr::select(
+      .data = data,
+      x = !!rlang::enquo(x),
+      y = !!rlang::enquo(y),
+      id.variable = !!rlang::enquo(id.variable)
+    ) %>%
+    tidyr::drop_na(data = .) %>%
+    dplyr::mutate(.data = .,
+                  x = droplevels(as.factor(x)),
+                  id.variable = droplevels(as.factor(id.variable))) %>%
+    tibble::as_tibble(x = .) %>%
+    dplyr::mutate_if(is.character, as.factor)
+
+  # run the ANOVA
+  stats_df <-
+    ez::ezANOVA(
+      data = data,
+      dv = y,
+      wid = id.variable,
+      within = x,
+      detailed = TRUE,
+      return_aov = TRUE
+  )
+
+  # get effect estimate and construct CIs
+  #
+  if (effsize.type == "unbiased") {
+    effsize <- "omega"
+    effsize.text <- quote(omega^2)
+    effsize_df <- sjstats::omega_sq(stats_df$aov,
+                                    ci.lvl = conf.level,
+                                    partial = FALSE)
+    effsize.estimate = effsize_df[2,2]
+    effsize.LL = effsize_df$conf.low[2]
+    effsize.UL = effsize_df$conf.high[2]
+  } else if (effsize.type == "biased") {
+    effsize <- "eta"
+    effsize.text <- quote(eta["p"]^2)
+    effsize_df <- sjstats::eta_sq(stats_df$aov,
+                                  ci.lvl = conf.level,
+                                  partial = TRUE)
+    effsize.estimate = effsize_df[2,2]
+    effsize.LL = effsize_df$conf.low[2]
+    effsize.UL = effsize_df$conf.high[2]
+  } else {
```

Comment:
Why are we supporting generalized eta-squared here? 

Maybe we can only support eta- and omega-squared (and not gain an entire additional dependency only for this one effect size)?

Reply:
It looked like a useful measure according to literature but I was mainly hopeful that it could overcome some of the deficiencies in `sjsstats` no strong objection if we leave it out for now since it doesn't

Reply:
Okay, I'd leave `eta_G` out for now. And we can come back to it if we feel like this is actually a better option than the other two. 

Avoiding `MBESS` avoids `gsl`; failing to download `gsl` is why the `linux` builds are currently failing for this PR. 

Once you make that change, I will merge this to master.

Reply:
Just made the committ